### PR TITLE
Update prop types and simplify tooling

### PR DESCRIPTION
**Pull request by taion**
Fixes https://github.com/react-bootstrap/react-prop-types/issues/25

### CODE REVIEW

**Comment by taion** [comment]
These are the API changes:
- **Breaking:** Remove `keyOf` and `singlePropFrom`
- **Breaking:** Rename `mountable` to `componentOrElement`
- **Bugfix:** Pass through additional arguments to wrapped validators

I've also tweaked the validation failure messages to more closely follow the current upstream React ones.

Tooling-wise, I've upgraded to Babel 6 and dropped Karma entirely. This means that we end up hacking around the test fixtures for `componentOrElement`.

I've also removed coverage for the time being, and the release script is just a `prepublish` hook because there are no separate docs or Bower repos.

This sets us up for addressing https://fb.me/react-unknown-prop.

File: test/isRequiredForA11ySpec.js [hunk 1]
```diff
@@ -1,19 +0,0 @@
-import React from 'react';
-import isRequiredForA11y from '../src/isRequiredForA11y';
-
-describe('isRequiredForA11y', function() {
```
Comment (jquense) [inline test/isRequiredForA11ySpec.js:4]: why remove this one?

Comment (taion) [inline test/isRequiredForA11ySpec.js:4]: I didn't, I renamed the retained ones to `.test.js`. Got carried away.

Comment (jquense) [inline test/isRequiredForA11ySpec.js:4]: ah ok. carry on

**Comment by taion** [comment]
@jquense Okay with shipping this?

**Comment by jquense** [comment]
ltgm

**Pull request merged**